README.md

```diff
@@ -94,491 +58,639 @@ By default, push-related components in the sample app won't work on Android, bec
+presenceMessages = await channel.presence.get(
+    ably.RestPresenceParams(
+        clientId: 'clientId',
+    ),
+);
+
+// Get presence messages with specific Connection ID
+presenceMessages = await channel.presence.get(
+    ably.RestPresenceParams(
+        connectionId: 'connectionId',
+    ),
+);
+```
+
+#### Read REST presence history
+
+```dart
+// Get presence history with default parameters
+ably.PaginatedResult<ably.PresenceMessage> history = await channel.presence.history();
+
+// Get presence history with custom parameters
+ably.PaginatedResult<ably.PresenceMessage> filteredHistory = await channel.presence.history(
+    ably.RestHistoryParams(
+        direction: 'forwards',
+        limit: 10,
+    )
+);
+```
+
+### PaginatedResult handling
+
+#### Get items on current page
+
+```dart
+// Example PaginatedResult returned from channel history
+ably.PaginatedResult<ably.Message> paginatedResult = await channel.history(params);
+
+// Get list of items from result
+List<ably.Message> items = paginatedResult.items;
+```
+
+#### Get next page if available
+
+```dart
+// Example PaginatedResult returned from channel history
+ably.PaginatedResult<ably.Message> paginatedResult = await channel.history(params);
+
+// Check if next page is available
+bool hasNextPage = paginatedResult.hasNext();
+
+// Fetch next page if it's available
+if (hasNextPage) {    
+  paginatedResult = await paginatedResult.next();
+}
+```
+
+### Encryption
+
+#### Create CipherParams
+
+```dart
+String key = 'base64EncodedKey'; // Can also be an UInt8List
+CipherParams cipherParams = ably.Crypto.getDefaultParams(key: key);
+```
+
+#### Setup encryption on a channel
+
+```dart
+// For Realtime
+RealtimeChannelOptions realtimeChannelOptions = ably.RealtimeChannelOptions(cipherParams: cipherParams);
+RealtimeChannel channel = realtime.channels.get("channel-name");
+channel.setOptions(realtimeChannelOptions)
+
+// For REST
+RestChannelOptions restChannelOptions = ably.RestChannelOptions(cipherParams: cipherParams);
+RestChannel channel = rest.channels.get("channel-name");
+channel.setOptions(restChannelOptions)
+```
+
+### Handling push notifications
 
 See [PushNotifications.md](PushNotifications.md) for detailed information on using PN with this plugin.
 
-## Caveats
+## Resources
+
+- [Quickstart Guide](https://www.ably.com/docs/quick-start-guide?lang=flutter)
+- [Introducing the Ably Flutter plugin](https://www.ably.com/blog/ably-flutter-plugin) by [Srushtika](https://github.com/Srushtika) (Developer Advocate)
+- [Building a Realtime Cryptocurrency App with Flutter](https://www.ably.com/tutorials/realtime-cryptocurrency-app-flutter) by [pr-Mais](https://github.com/pr-Mais) and [escamoteur](https://github.com/escamoteur)
+- [Building realtime apps with Flutter and WebSockets: client-side considerations](https://www.ably.com/topic/websockets-flutter)
+
+## Requirements
+
+### Flutter
+
+Flutter 2.5.0 or higher is required.
+
+### iOS
```

Comment:
Looking at this header with fresh eyes (I acknowledge you've simply moved it), I wonder if it might be expanded:

```suggestion
### iOS Runtime Requirements
```

This'll make the deep link into this section of this document make more sense for those that share it (will be `#ios-runtime-requirements`), plus it makes it more explicit that we're talking about the end-user requirements.

Same goes for the 'Android' header below.

Reply:
Fixed in https://github.com/ably/ably-flutter/pull/413/commits/6175af9b8543229dd646734e80da9162bb16a895